src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs

```diff
@@ -383,11 +383,21 @@ public override Task<byte[]> GetArchiveHashAsync(HashAlgorithmName hashAlgorithm
 
         public override bool CanVerifySignedPackages(SignedPackageVerifierSettings verifierSettings)
         {
+            var value = false;
 #if IS_DESKTOP
-            return true;
-#else
-            return false;
+            if (RuntimeEnvironmentHelper.IsMono) {
+                if (RuntimeEnvironmentHelper.IsWindows) {
```

Comment:
Not sure if we care about the mono on Windows scenario here. @dtivel Is that scenario relevant?

Did you test that the RuntimeEnvironmentHelper actually returns true for running under windows?

Reply:
@nkolev92 I added a test for RuntimeEnvironmentHelper. I did some manual testing and it does return true for IsWindows when you're running on Windows; false otherwise.